src/main/java/com/amazonaws/codedeploy/AWSCodeDeployPublisher.java

```diff
@@ -279,9 +283,25 @@ private void verifyCodeDeployApplication(AWSClients aws) throws IllegalArgumentE
         }
     }
 
-    private RevisionLocation zipAndUpload(AWSClients aws, String projectName, FilePath sourceDirectory) throws IOException, InterruptedException, IllegalArgumentException {
+    private RevisionLocation compressAndUpload(AWSClients aws, String projectName, FilePath sourceDirectory, String fileType) throws IOException, InterruptedException, IllegalArgumentException {
 
-        File zipFile = File.createTempFile(projectName + "-", ".zip");
+		String extension;
+		BundleType bundleType;
+		if (fileType == null || fileType.equals("Tar")) {
+        	extension = ".zip";
+            bundleType = BundleType.Zip;
+        } else if (fileType.equals("Tar")) {
+            extension = ".tar";
+            bundleType = BundleType.Tar;
+        } else if (fileType.equals("Tgz")) {
+            extension = ".tar.gz";
+            bundleType = BundleType.Tgz;
+        } else {
```

Comment:
I'd rather we throw on an unknown file type since you're already handling the default null case (maybe update it to include empty string too)